src/authorization/AuxiliaryReader.ts

```diff
@@ -1,56 +1,59 @@
+  private readonly reader: PermissionReader;
   private readonly auxiliaryStrategy: AuxiliaryStrategy;
 
-  public constructor(resourceReader: PermissionReader, auxiliaryStrategy: AuxiliaryStrategy) {
+  public constructor(reader: PermissionReader, auxiliaryStrategy: AuxiliaryStrategy) {
     super();
-    this.resourceReader = resourceReader;
+    this.reader = reader;
     this.auxiliaryStrategy = auxiliaryStrategy;
   }
 
-  public async canHandle(auxiliaryAuth: PermissionReaderInput): Promise<void> {
-    const resourceAuth = this.getRequiredAuthorization(auxiliaryAuth);
-    return this.resourceReader.canHandle(resourceAuth);
-  }
-
-  public async handle(auxiliaryAuth: PermissionReaderInput): Promise<PermissionSet> {
-    const resourceAuth = this.getRequiredAuthorization(auxiliaryAuth);
-    this.logger.debug(`Checking auth request for ${auxiliaryAuth.identifier.path} on ${resourceAuth.identifier.path}`);
-    return this.resourceReader.handle(resourceAuth);
-  }
+  public async handle({ requestedModes, credentials }: PermissionReaderInput): Promise<PermissionMap> {
+    const auxiliaries = this.findAuxiliaries(requestedModes);
 
-  public async handleSafe(auxiliaryAuth: PermissionReaderInput): Promise<PermissionSet> {
-    const resourceAuth = this.getRequiredAuthorization(auxiliaryAuth);
-    this.logger.debug(`Checking auth request for ${auxiliaryAuth.identifier.path} to ${resourceAuth.identifier.path}`);
-    return this.resourceReader.handleSafe(resourceAuth);
-  }
+    const updatedMap = modify(new IdentifierSetMultiMap(requestedModes),
+      { add: auxiliaries.values(), remove: auxiliaries.keys() });
+    const result = await this.reader.handleSafe({ requestedModes: updatedMap, credentials });
 
-  private getRequiredAuthorization(auxiliaryAuth: PermissionReaderInput): PermissionReaderInput {
-    if (!this.auxiliaryStrategy.isAuxiliaryIdentifier(auxiliaryAuth.identifier)) {
-      throw new NotImplementedHttpError('AuxiliaryAuthorizer only supports auxiliary resources.');
+    for (const [ identifier, [ subject ]] of auxiliaries) {
+      this.logger.debug(`Mapping ${subject.path} permissions to ${identifier.path}`);
+      result.set(identifier, result.get(subject) ?? {});
     }
+    return result;
+  }
 
-    if (this.auxiliaryStrategy.usesOwnAuthorization(auxiliaryAuth.identifier)) {
-      throw new NotImplementedHttpError('Auxiliary resource uses its own permissions.');
+  /**
+   * Maps auxiliary resources that do not have their own authorization checks to their subject resource.
+   */
+  private findAuxiliaries(requestedModes: AccessMap): IdentifierMap<[ResourceIdentifier, ReadonlySet<AccessMode>]> {
+    const auxiliaries = new IdentifierMap<[ResourceIdentifier, ReadonlySet<AccessMode>]>();
+    for (const [ identifier, modes ] of requestedModes.entrySets()) {
+      if (this.isAuxiliary(identifier)) {
+        auxiliaries.set(identifier, [ this.auxiliaryStrategy.getSubjectIdentifier(identifier), modes ]);
```

Comment:
The reason for using the array here and in the other cases is because I can then feed it directly into a Map constructor without needing an additional `map` step.

Reply:
Aah they are entries? Can we then maybe just type it `IdentifierMap<Entry<ResourceIdentifier, ReadonlySet<AccessMode>>` with `Entry<X, Y extends Iterable> = [X, Y]` or am I wrong / is there no added value?

Reply:
I am not complaining for esthetics or anything; it's that those bits were hard to understand for me.

Alternatively, we could still go for `{ subject, modes }` and a `mapFrom` helper or so at a later stage.
There's a several such arrays that require context for me.

Reply:
> Aah they are entries? Can we then maybe just type it `IdentifierMap<Entry<ResourceIdentifier, ReadonlySet<AccessMode>>` with `Entry<X, Y extends Iterable> = [X, Y]` or am I wrong / is there no added value?

We can do better actually. In this PR I also introduce a `MapEntry` type so we can do `IdentifierMap<MapEntry<AccessMap>>` which I think would make it clear?

Reply:
That's the one!